coral-presto/src/test/java/com/linkedin/coral/presto/rel2presto/HiveToPrestoConverterTest.java

```diff
@@ -1,16 +1,87 @@
 /**
- * Copyright 2017-2020 LinkedIn Corporation. All rights reserved.
+ * Copyright 2017-2021 LinkedIn Corporation. All rights reserved.
  * Licensed under the BSD-2 Clause license.
  * See LICENSE in the project root for license information.
  */
 package com.linkedin.coral.presto.rel2presto;
 
+import org.apache.calcite.rel.RelNode;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import static org.testng.Assert.*;
+
 
 public class HiveToPrestoConverterTest {
 
-  @Test
-  public void basicSql() {
+  RelToPrestoConverter rel2Presto;
+
+  @BeforeTest
+  public void beforeClass() throws HiveException, MetaException {
+    TestUtils.initializeViews();
+    rel2Presto = new RelToPrestoConverter();
+  }
+
+  @Test(dataProvider = "viewTestCases")
+  public void testViews(String database, String view, String expectedSql) {
+    RelNode relNode = TestUtils.convertView(database, view);
+    String expandedSql = rel2Presto.convert(relNode);
+    assertTrue(expandedSql.contains(expectedSql));
+  }
+
+  @DataProvider(name = "viewTestCases")
+  public Object[][] viewTestCasesProvider() {
+    return new Object[][] { { "test", "fuzzy_union_view", "" + "SELECT \"a\", \"b\"\n" + "FROM (SELECT \"a\", \"b\"\n"
+        + "FROM \"test\".\"tablea\"\n" + "UNION ALL\n" + "SELECT \"a\", \"b\"\n" + "FROM \"test\".\"tablea\")" },
```

Comment:
What is the reason those strings are formatted this way? Were there new lines originally after the `+` signs then spotless removed them?

Reply:
yeah. spotless enforced the format. No way we can control that right now.

Reply:
Could you remove unnecessary `+` signs, `\n`, and `""`?

Reply:
To be clear, 1. removing + signs will not make the situation getting better due to the spotless reformatting issue. You will see a long string. 2. \n cannot be removed because it is included in the parsed result. 3. The only thing we can remove is the leading "" which is introduced by Raymond. Not quite sure why it was approved for checking in.

Reply:
I tried to make it to be standardized based on the standard which I use normally.